dotnet/samples/Concepts/Agents/ChatCompletion_ServiceSelection.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace Agents;
+
+/// <summary>
+/// Demonstrate service selection for <see cref="ChatCompletionAgent"/> through setting service-id
+/// on <see cref="ChatCompletionAgent.Arguments"/> and also providing override <see cref="KernelArguments"/>
+/// when calling <see cref="ChatCompletionAgent.InvokeAsync"/>
+/// </summary>
+public class ChatCompletion_ServiceSelection(ITestOutputHelper output) : BaseTest(output)
+{
+    private const string ServiceKeyGood = "chat-good";
+    private const string ServiceKeyBad = "chat-bad";
+
+    [Fact]
+    public async Task UseServiceSelectionWithChatCompletionAgentAsync()
+    {
+        // Create kernel with two instances of IChatCompletionService
+        // One service is configured with a valid API key and the other with an
+        // invalid key that will result in a 401 Unauthorized error.
+        Kernel kernel = CreateKernelWithTwoServices();
+
+        // Define the agent targeting ServiceId = ServiceKeyGood 
+        ChatCompletionAgent agentGood =
+            new()
+            {
+                Kernel = kernel,
+                Arguments = new(new OpenAIPromptExecutionSettings() { ServiceId = ServiceKeyGood }),
+            };
+
+        // Define the agent targeting ServiceId = ServiceKeyGood 
+        ChatCompletionAgent agentBad =
+            new()
+            {
+                Kernel = kernel,
+                Arguments = new(new OpenAIPromptExecutionSettings() { ServiceId = ServiceKeyBad }),
+            };
+
+        // Define the agent with no explicit ServiceId defined
+        ChatCompletionAgent agentDefault = new() { Kernel = kernel };
+
+        // Invoke agent as initialized with ServiceId = ServiceKeyGood
```

Comment:
I think it would be helpful if the comment described what is going to happen so that readers can understand difference in result with these different Invokes without running the code. 

Maybe something like: 
`// Invoke agent as initialized with ServiceId = ServiceKeyGood - Should work as expected`...
`// Invoke agent as initialized with ServiceId = ServiceKeyBad - Will fail due to bad service key`...
`// Invoke agent with override arguments where ServiceId = ServiceKeyGood - Works`

Reply:
Updated